### ISSUE TITLE
Sync database with drop not working

### ISSUE THREAD

**Issue by techtastet** [eveningkid/denodb#216]
When i call my db with : `await db.sync({drop: true});`

since I've been using denodb, I've never been able to properly drop my tables with relations

The logs :
```shell
INFO connecting localhost:8889
INFO connected to localhost:8889
INFO close connection
error: Uncaught (in promise) Error: Cannot delete or update a parent row: a foreign key constraint fails
      throw new Error(error.message);
            ^
    at PoolConnection.nextPacket (https://deno.land/x/mysql@v2.8.0/src/connection.ts:185:13)
    at async PoolConnection.execute (https://deno.land/x/mysql@v2.8.0/src/connection.ts:263:21)
    at async https://deno.land/x/mysql@v2.8.0/src/client.ts:97:14
    at async Client.useConnection (https://deno.land/x/mysql@v2.8.0/src/client.ts:107:14)
    at async Client.execute (https://deno.land/x/mysql@v2.8.0/src/client.ts:96:12)
    at async MySQLConnector.query (https://deno.land/x/denodb@v1.0.24/lib/connectors/mysql-connector.ts:71:22)
    at async Database.query (https://deno.land/x/denodb@v1.0.24/lib/database.ts:226:21)
    at async Function.drop (https://deno.land/x/denodb@v1.0.24/lib/model.ts:137:5)
    at async Database.sync (https://deno.land/x/denodb@v1.0.24/lib/database.ts:191:9)

```

**Comment by techtastet** [comment]
I tested the queries on my side and the problem seems to be in the order of deleting the tables.
In a specific order, the constraints are no longer imposed and the tables are deleted.

**Comment by techtastet** [comment]
The other should be reversed with respect to the models at initialization.

Because when we instantiate models we have to pay attention to the order in order not to have relations that execute badly.

It's the same here but with the order in `.reverse()`

**Comment by eveningkid** [comment]
Thanks again Thomas.

Just replied on the PR, hope we can merge this soon :)

**Comment by eveningkid** [comment]
Fix is now available through `v1.0.25`
